File: LeetCode/Math/Golang/Leetcode795-number-of-subarrays-with-bounded-maximum.go
package sbgo

func numSubarrayBoundedMax(nums []int, left int, right int) int {
	n := len(nums)
	calcu := func(mx int) int {
		res := 0
		for i := 0; i < n; i++ {
			if nums[i] > mx {
				continue
			}
			j := i + 1
			for j < n && nums[j] <= mx {
				j++
			}
			dis := j - i
			res += dis * (dis + 1) / 2
			i = j
		}
		return res
	}
	return calcu(right) - calcu(left-1)
}

File: LeetCode/Math/JS/Leetcode795-number-of-subarrays-with-bounded-maximum.js
var numSubarrayBoundedMax = function (nums, left, right) {
  const n = nums.length;
  let calcu = (mx) => {
    let res = 0;
    for (let i = 0; i < n; i++) {
      if (nums[i] > mx) continue;
      let j = i + 1;
      while (j < n && nums[j] <= mx) j++;
      const dis = j - i;
      res += Math.floor((dis + 1) * dis / 2);
      i = j;
    }
    return res;
  }
  return calcu(right) - calcu(left - 1);
};